vms/administrator/forms.py

```diff
@@ -16,6 +18,61 @@ class Meta:
             'phone_number', 'email'
         ]
 
+    def check_if_google_email(self):
+        gmail_host_pattern = re.compile("(.google.com.|.googlemail.com.|.psmtp.com.)$", re.IGNORECASE)
+        data = self.cleaned_data['email']
+        test_data = data.slice('@')
+        
+        if test_data and test_data!=" ":
+            host = test_data[-1]
+        else:
+            return False
+
+        if host == "gmail.com" or host == "googlemail.com" or host == "psmtp.com":
```

Comment:
I think you misunderstood the issue; we have to check for duplicate entries in *our database.*
We simply need to check that the email entered by a user should not be present in our database.
I don't really get the need of host pattern and all. Kindly correct me if I'm wrong.

Reply:
@Harirai  , there are two functions. The first one, check_if_google_email is a check if the email belongs to gmail and also as an extra layer of validation checks if the host of the email is valid or not. The second function, check_duplication tries to remove  all periods and + symbols from the email and then checks in the database of the VMS if that email already exists or not. So that's how i have attempted to solve he issue if duplication. If still there is an element of doubt, please ping me again and I'll try to clarify it. Thanks!